src/Symfony/Component/Routing/Loader/Configurator/Traits/RouteTrait.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Loader\Configurator\Traits;
+
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+
+trait RouteTrait
+{
+    /**
+     * @var RouteCollection|Route
+     */
+    private $route;
+
+    /**
+     * Adds defaults.
+     *
+     * @param array $defaults
+     *
+     * @return $this
+     */
+    final public function defaults(array $defaults)
+    {
+        $this->route->addDefaults($defaults);
+
+        return $this;
+    }
+
+    /**
+     * Adds requirements.
+     *
+     * @param array $requirements
+     *
+     * @return $this
+     */
+    final public function requirements(array $requirements)
+    {
+        $this->route->addRequirements($requirements);
+
+        return $this;
+    }
+
+    /**
+     * Adds options.
+     *
+     * @param array $options
+     *
+     * @return $this
+     */
+    final public function options(array $options)
+    {
+        $this->route->addOptions($options);
+
+        return $this;
+    }
+
+    /**
+     * Sets the condition.
+     *
+     * @param string $condition
+     *
+     * @return $this
+     */
+    final public function condition($condition)
+    {
+        $this->route->setCondition($condition);
+
+        return $this;
+    }
+
+    /**
+     * Sets the pattern for the host.
+     *
+     * @param string $pattern
+     *
+     * @return $this
+     */
+    final public function host($pattern)
+    {
+        $this->route->setHost($pattern);
+
+        return $this;
+    }
+
+    /**
+     * Sets the schemes (e.g. 'https') this route is restricted to.
+     * So an empty array means that any scheme is allowed.
+     *
+     * @param array $schemes
+     *
+     * @return $this
+     */
+    final public function schemes(array $schemes)
+    {
+        $this->route->setSchemes($schemes);
+
+        return $this;
+    }
+
+    /**
+     * Sets the HTTP methods (e.g. 'POST') this route is restricted to.
+     * So an empty array means that any method is allowed.
+     *
+     * @param array $methods
+     *
+     * @return $this
+     */
+    final public function methods(array $methods)
+    {
+        $this->route->setMethods($methods);
+
+        return $this;
+    }
+
+    /**
+     * Adds the "_controller" entry to defaults.
+     *
+     * @param callable $controller a callable or parseable pseudo-callable
+     *
+     * @return $this
+     */
+    final public function controller($controller)
```

Comment:
I had an idea looking at the examples on the main blog post about this feature (which looks great btw): http://symfony.com/blog/new-in-symfony-php-based-configuration-for-services-and-routes

Is it possible, or does it make sense to change the api of controllers (or add a new helper method), to enable use `Controller::class`, without having to concatenate the results?

If I'm not mistaken, all other class methods accept a class as it's parameter, so you can pass a string, or use the `::class` (which I find less error prone).

This one however, since it's following the config that already exist would end in being something like:

```php
->add('homepage', '/')
    ->controller('App\Controller\DefaultController::index')
```
 or

```
->add('homepage', '/')
    ->controller(DefaultController::class . '::index')
```

In the later scenario, would changing this method's parameters or adding a new method that would accept the method of the controller separately help, as far as DX goes?

as an example:
```php
->add('homepage', '/')
    ->controller(DefaultController::class, 'index')
```

I understand that wouldn't match the full string config for a controller, in relation to other configurations, so that might be a con, and concatenating honestly doesn't look that bad either, just wanted to raise the case, since it's different from all others, in case it makes sense.

Reply:
@ajessu That's indeed a very good idea 👍 of course this action parameter should be optional as more advanced developers use a single action per controller (called Action controllers).

Reply:
that's already possible: just use the array-callable PHP syntax:
`->controller([DefaultController::class, 'index'])`

Reply:
Thanks @nicolas-grekas. That seems perfect. Wasn't aware that was a possibility here.